src/app/qgisapp.cpp

```diff
@@ -7693,21 +7693,8 @@ void QgisApp::duplicateLayers( const QList<QgsMapLayer *>& lyrList )
         }
 
         //add variables defined in layer properties
-        QStringList variableNames = vlayer->customProperty( "variableNames" ).toStringList();
-        QStringList variableValues = vlayer->customProperty( "variableValues" ).toStringList();
-
-        int varIndex = 0;
-        Q_FOREACH ( const QString& variableName, variableNames )
-        {
-          if ( varIndex >= variableValues.length() )
-          {
-            break;
-          }
-
-          QVariant varValue = variableValues.at( varIndex );
-          varIndex++;
-          QgsExpressionContextUtils::setLayerVariable( dupVLayer, variableName, varValue );
-        }
+        QVariantMap vars = vlayer->customProperty( "expressionVariables" ).toMap();
+        dupVLayer->setCustomProperty( "expressionVariables", vars );
```

Comment:
Shouldn't "duplicateLayers" copy ALL custom properties? This looks really fragile